addons/SmoothScroll/SmoothScrollContainer.gd

```diff
@@ -634,11 +634,16 @@ func scroll_to_fauna(duration:float=0.5) -> void:
 	if(!PlayerManager.is_player_initial_position_set):
 		return
 	
-	var room_distance: float = 66
-	var scroll_to_pos:float = -room_distance
-	var rooms_between_fauna_and_top : int = MapManager.map_width_array.size() - 8
+	var map_UI : MapUI = get_parent()
+	var current_map_texure_button : TextureButton = Helpers.get_first_child_node_of_type(map_UI.current_player_room, TextureButton)
```

Comment:
I tested and  `var current_map_texure_button : TextureButton = map_UI.current_player_room.button;` also works, it's more explicit and there won't be a problem if we have multiple texture buttons on this later on

Reply:
Thanks, didn't know about that variable.

Regarding the scrolling, that is a built in thing to scroll smoother, so I can change that by just by giving a parameter

Reply:
the problem isn't how hard it scrolls, the problem is that we see the movement
even if you set the smoothness to 0, if we spawn the map directly I'm not sure the display won't feel like it's jumping to the position, thus why i said we should add a little delay
if it works without the delay, then good
but we also need the smooth scroll after the map loaded for the player

Reply:
I don't see any problems like you're describing, at least not after the changes I made, so I hope it's fine

Reply:
Mmh I'm seeing a little quick jump around floors 3 or 4, but it's not very noticeable
It might become more noticeable if we add more textures, or maybe it won't i'm not sure
Let's keep it like that for now